benches/read.rs

```diff
@@ -60,6 +60,24 @@ fn read_single_image_rle_all_channels(bench: &mut Bencher) {
     })
 }
 
+/// Read without multi-core RLE decompression
+fn read_single_image_rle_non_parallel_all_channels(bench: &mut Bencher) {
+    bench.iter(||{
+        let path = "tests/images/valid/custom/crowskull/crow_rle.exr";
```

Comment:
Is it desired that the file is loaded from disk in each iteration? To avoid unnecessary wait times, I would suggest first loading the bytes from memory, then afterwards decode the image from those in-memory file bytes. We should probably change all benchmarks to use this strategy. You can see it in line 26, in the function `read_single_image_uncompressed_from_buffer_rgba`. 

This also makes sure that file access does not intruduce noise into the benchmark

Reply:
Makes sense. The inputs also have to be `black_box`ed in this case. Also `std::hint::black_box` was stabilized recently, it's probably a good idea to use it.

More info black boxing: [Why my Rust benchmarks were wrong, or how to correctly use std::hint::black_box?](https://gendignoux.com/blog/2022/01/31/rust-benchmarks.html)

Reply:
I woud appreciate if you did that for all of the benchmarks. Including the black boxing of the input :)

Reply:
I've added from-memory and from-disk benchmarks for comparison. Not sure if they're useful at all, but I guess they validate that the from-disk one doesn't do anything stupid.

I've also split reading with format conversion into a separate file and explicitly labeled f32_to_f32 and f32_to_f16 conversions, and also added an f32_to_u32 benchmark.

Reply:
> I would appreciate if you did that for all of the benchmarks. Including the black boxing of the input :)

I don't want to pile even more things into this PR. Let's get this merged and I'll do a benchmark refactoring PR separately.

Reply:
Great work! Thanks :) I'm curious to see how much buffering will speed up the decoding